### ISSUE TITLE
GraphQL API doesn't update optional values

### ISSUE THREAD

**Issue by chris-mds** [aws-amplify/amplify-ios#800]
**Describe the bug**
Im trying to implement AppSync API and it doesn't update optional Values, when updating on the client app side.

**To Reproduce**
1. Create a DataModel with optional fields
2. Create on your app an item of this model with any optional key filled with value 'nil'
3. Now we can see in our Table that this key is from type 'NULL' and his value is true.
4. Now update on your app this key with the value it should be, when it is not nil. E.g. string.
5. The Xcode console will show that the operation was successfull and the item was updated. But in the success data you can see, that this value wasn't updated
6. Refresh the table and you will see, that nothing happened. The key is still from type 'NULL' and set to true

**Expected behavior**
Update the key with its given new value

**Screenshots**
<img width="437" alt="Bildschirmfoto 2020-10-04 um 16 45 59" src="https://user-images.githubusercontent.com/71610076/95018732-d6300e00-0661-11eb-984f-27abb52aab0a.png">
<img width="485" alt="Bildschirmfoto 2020-10-04 um 16 45 20" src="https://user-images.githubusercontent.com/71610076/95018736-db8d5880-0661-11eb-9778-a250f5b7b5eb.png">
<img width="494" alt="Bildschirmfoto 2020-10-04 um 16 46 17" src="https://user-images.githubusercontent.com/71610076/95018737-dd571c00-0661-11eb-8353-a74e9a7695e2.png">

**Environment(please complete the following information):**
 - Dependency Manager: [ Cocoapods]
 - Swift Version : [ 5.3]

**Device Information (please complete the following information):**
 - Device: [Simulator]
 - iOS Version: [iOS 14]

**Comment by lawmicha** [comment]
Hi @chris-mds 

Sorry you are running into this issue, can you please verify which Amplify pod versions you are using?

I was not able to reproduce using Amplify 1.3.1 on Xcode 12 

```
let post = Post(title: "title", content: "content", createdAt: .now(), updatedAt: nil, draft: nil, rating: nil, status: nil, finishedFrom: nil, isFinished: nil, finishedAt: nil, comments: [])
                Amplify.API.mutate(request: .create(post)) { (result) in
                    switch result {
                    case .success(let result):
                        print(result)
                    case .failure(let apiError):
                        print(apiError)
                    }
                }
```
<img src="https://user-images.githubusercontent.com/1365977/95816603-5b579a80-0cd4-11eb-9bbc-8118adb13da1.png" width="400">


```
_ = Amplify.API.query(request: .get(Post.self, byId: "BF95FF22-8620-4758-B048-D843BFCEB8E8")) { (result) in
                    switch result {
                    case .success(let result):
                        print(result)
                        switch result {
                        case .success(let post):
                            if var post = post {
                                post.finishedFrom = "Chris"
                                post.isFinished = true
                                post.comments = []
                                Amplify.API.mutate(request: .update(post)) { (result) in
                                    switch result {
                                    case .success(let result):
                                        print(result)
                                    case .failure(let apiError):
                                        print(apiError)
                                    }
                                }
                            }
                        case .failure(let graphQLError):
                            print(graphQLError)
                        }
                        
                    case .failure(let apiError):
                        print(apiError)
                    }
                }
```
<img src="https://user-images.githubusercontent.com/1365977/95816638-6e6a6a80-0cd4-11eb-9fad-538809986e2a.png" width="400">


There was a known issue fixed in 1.2.0 and later

**Comment by chris-mds** [comment]
Since I was mixing APIPlugin and DataStorePlugin and I haven't handled versions, the conflict detection rejected any updates. Now with a clean API and no mixes it works like expected, thanks!
